tmt/convert.py

```diff
@@ -46,6 +46,116 @@ def repr_str(dumper, data):
 #  Convert
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+def read_manual(plan_id, case_id, disabled):
+    """
+    Reads metadata of manual test cases from Nitrate
+    """
+    try:
+        import nitrate
+    except ImportError:
+        raise ConvertError('Install nitrate module to import metadata.')
+    # Turns off nitrate caching
+    nitrate.set_cache_level(0)
+
+    try:
+        tree = fmf.Tree(os.getcwd())
+    except fmf.utils.RootError:
+        raise ConvertError("Initialize metadata tree using 'tmt init'")
+
+    try:
+        if plan_id:
+            all_cases = nitrate.TestPlan(int(plan_id)).testcases
+            case_ids = [case.id for case in all_cases if case.manual]
+        else:
+            case_ids = [int(case_id)]
+    except ValueError:
+        raise ConvertError('Value of plan/case option must be an integer')
+
+    # Create directory to store manual tests in
+    old_cwd = os.getcwd()
+    os.chdir(tree.root)
+    try:
+        os.mkdir('Manual')
+    except FileExistsError:
+        pass
+
+    os.chdir('Manual')
+
+    for case_id in case_ids:
+        testcase = nitrate.TestCase(case_id)
+        if testcase.status.name != 'CONFIRMED' and not disabled:
+            continue
+
+        # Filename sanitization
+        dir_name = testcase.summary.replace(' ', '_')
+        dir_name = dir_name.replace('/', '_')
+        try:
+            os.mkdir(dir_name)
+        except FileExistsError:
+            pass
+
+        os.chdir(dir_name)
+        echo("Importing the '{0}' test case.".format(dir_name))
+
+        # Test case data
+        md_content = {}
+        md_content['Setup'] = testcase.setup
+        md_content['Actions'] = testcase.action
+        md_content['Breakdown'] = testcase.breakdown
+        md_content['Expected Results'] = testcase.effect
```

Comment:
Please, keep the naming defined in the [specification](https://tmt.readthedocs.io/en/latest/spec/tests.html#manual):

```markdown
# Setup
Prepare the env.

# Test

## Step
Do this and that.

## Expect
Check this and that.

# Cleanup
Cleanup the env.
```

Note that steps and expected results should be the second level heading.

Reply:
As far as I can see there is nothing in specification about mapping existing format of manual tests to this new format.
The issue is that new specification has only 3 main sections (Setup, Test, Cleanup) to 4 sections of old one (Setup, Action, Breakdown, Expected result).
Mapping Setup-Setup is obvious however the rest is quite problematic in my opinion.
I could map old Action and Expected result under new Test section (Action would be second level Step and Expected Result would be second level Expect), but I am not sure it would work for most tests.
Cleanup would be empty.
I have no idea what to do with Breakdown though.

Any ideas?

Reply:
I'd suggest the following mapping:

* setup ... setup
* action ... test/step
* expected result ... test/expect
* breakdown ... cleanup

We don't have to cover all possible corner cases. Users can adjust content after import.

Reply:
Changed in d65b862
However in cases I tried to import the Cleanup setion (original Breakdown) does not contain cleanup instruction as I feared. It will have to be manually changed or it won't make much sense.

Reply:
Thanks for fixing this.